### PR TITLE
Update Simple Upscale Button to work with spandrel models

### PR DESCRIPTION
**Pull request by chainchompa**
## Summary
Update Simple Upscale Button to work with spandrel models, add UpscaleWarning when models aren't available, clean up ESRGAN logic

## Checklist

- [ ] _The PR has a short but descriptive title, suitable for a changelog_
- [ ] _Tests added / updated (if applicable)_
- [ ] _Documentation added / updated (if applicable)_

### CODE REVIEW

File: invokeai/frontend/web/src/features/nodes/util/graph/buildAdHocUpscaleGraph.ts [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
+      [SPANDREL]: upscaleNode,
     },
     edges: [],
   };
 
-  addCoreMetadataNode(graph, {}, ESRGAN);
+  addCoreMetadataNode(graph, {}, SPANDREL);
   upsertMetadata(graph, {
-    esrgan_model: esrganModelName,
+    spandrel_model: simpleUpscaleModel,
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/nodes/util/graph/buildAdHocUpscaleGraph.ts:36]:
```suggestion
    upscale_model: simpleUpscaleModel,
```

File: invokeai/frontend/web/src/features/parameters/components/Upscale/ParamSimpleUpscale.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { Combobox, FormControl, FormLabel } from '@invoke-ai/ui-library';
+import { useAppDispatch, useAppSelector } from 'app/store/storeHooks';
+import { useModelCombobox } from 'common/hooks/useModelCombobox';
+import { simpleUpscaleModelChanged } from 'features/parameters/store/upscaleSlice';
+import { memo, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSpandrelImageToImageModels } from 'services/api/hooks/modelsByType';
+import type { SpandrelImageToImageModelConfig } from 'services/api/types';
+
+const ParamSimpleUpscale = () => {
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/parameters/components/Upscale/ParamSimpleUpscale.tsx:10]: I would recommend reusing `ParamSpandrelModel`. You could add a prop for which part of the uspcale state it should be using/updating

File: invokeai/frontend/web/src/features/nodes/util/graph/buildAdHocUpscaleGraph.ts (hunk 1 above)
Comment (maryhipp) [reply invokeai/frontend/web/src/features/nodes/util/graph/buildAdHocUpscaleGraph.ts:36]: Most place where we build model metadata, we use this `getModelMetadataField` util. I'm not sure what the difference is but I'd look into using that here.

File: invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/modelsLoaded.ts [hunk 3]
```diff
@@ -204,3 +205,23 @@ const handleSpandrelImageToImageModels: ModelHandler = (models, state, dispatch,
 
   dispatch(upscaleModelChanged(null));
 };
+
+const handleSimpleUpscaleModels: ModelHandler = (models, state, dispatch, _log) => {
```
Comment (maryhipp) [inline invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/modelsLoaded.ts:209]: I would just put this log in the `handleSpandrelImageToImageModels` listener

Comment (chainchompa) [reply invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/modelsLoaded.ts:209]: ModelHandler requires a log, or do you mean just adding all the handleSimpleUpscaleModels logic to handleSpandrelImageToImageModels?

**Pull request merged**